packages/sdk/src/util/gridLayout.ts

```diff
@@ -0,0 +1,182 @@
+	 */
+	public constructor(
+		private anchor: Actor,
+		public gridAlignment = BoxAlignment.MiddleCenter,
+		public defaultCellAlignment = BoxAlignment.MiddleCenter
+	) { }
+
+	/** The number of columns in this grid. */
+	public columnCount() {
+		return this.contents.map(c => c.column).reduce(max, -1) + 1;
+	}
+
+	/** The number of rows in this grid. */
+	public rowCount() {
+		return this.contents.map(c => c.row).reduce(max, -1) + 1;
+	}
+
+	/** The width of the full grid. */
+	public gridWidth() {
+		const colCount = this.columnCount();
+		let width = 0;
+		for (let i = 0; i < colCount; i++) {
+			width += this.columnWidth(i);
+		}
+		return width;
+	}
+
+	/** The height of the full grid. */
+	public gridHeight() {
+		const rowCount = this.rowCount();
+		let height = 0;
+		for (let i = 0; i < rowCount; i++) {
+			height += this.rowHeight(i);
+		}
+		return height;
+	}
+
+	/**
+	 * The width of a particular column.
+	 * @param i The column index.
+	 */
+	public columnWidth(i: number) {
+		return this.contents.filter(c => c.column === i).map(c => c.width).reduce(max, 0);
+	}
+
+	/**
+	 * The height of a particular row.
+	 * @param i The row index.
+	 */
+	public rowHeight(i: number) {
+		return this.contents.filter(c => c.row === i).map(c => c.height).reduce(max, 0);
+	}
+
+	/** The widths of every column. */
+	public columnWidths() {
+		return this.contents.reduce((arr, c) => {
+			arr[c.column] = Math.max(arr[c.column] ?? 0, c.width);
+			return arr;
+		}, [] as number[]);
+	}
+
+	/** The heights of every row. */
+	public rowHeights() {
+		return this.contents.reduce((arr, c) => {
+			arr[c.row] = Math.max(arr[c.row] ?? 0, c.height);
+			return arr;
+		}, [] as number[]);
+	}
+
+	/**
+	 * Add an actor to the grid. The actor's position will not be updated until [[applyLayout]] is called.
+	 * @param options The cell's configuration.
+	 */
+	public addCell(options: AddCellOptions) {
+		const { contents } = options;
+
+		if (contents.parent !== this.anchor) {
+			throw new Error("Grid cell contents must be parented to the grid root");
+		}
+
+		// insert cell
+		this.contents.push(options);
+	}
+
+	/** Recompute the positions of all actors in the grid. */
+	public applyLayout() {
+		const colWidths = this.columnWidths();
+		const rowHeights = this.rowHeights();
+		const gridAlign = GridLayout.getOffsetFromAlignment(
+			InvertBoxAlignment(this.gridAlignment),
+			colWidths.reduce(sum, 0),
+			rowHeights.reduce(sum, 0))
+			.negate();
+
+		for (const cell of this.contents) {
+			const cellPosition = new Vector3(
+				colWidths.slice(0, cell.column).reduce(sum, 0),
+				-rowHeights.slice(0, cell.row).reduce(sum, 0),
+				cell.contents.transform.local.position.z);
+			const cellAlign = GridLayout.getOffsetFromAlignment(
+				cell.alignment ?? this.defaultCellAlignment,
+				colWidths[cell.column], rowHeights[cell.row]
+			);
+
+			cell.contents.transform.local.position = gridAlign.add(cellPosition).add(cellAlign);
```

Comment:
>cell.contents.transform.local.position [](start = 3, length = 38)

Would be nice: Option to interpolate this change.

Reply:
Easy to add, done!